services/beaconchain/src/main/java/tech/pegasys/teku/services/beaconchain/BeaconChainController.java

```diff
@@ -981,6 +981,7 @@ protected void initP2PNetwork() {
             .keyValueStore(keyValueStore)
             .requiredCheckpoint(weakSubjectivityValidator.getWSCheckpoint())
             .specProvider(spec)
+            .recordMessageArrival(true)
```

Comment:
I don't get the benefits of setting this flag at P2PNetwork level, since it is hardcoded to be true.
If move this flag into `Eth2TopicHandler` constructor, than we can activate the recording per topic (in this case we can activate it only for blocks) and let  the messageFactory in `LibP2PGossipNetworkBuilder.createGossipRouter` dynamically chose base on topic handler if we want to get current time or not. Otherwise, for simplicity, we could always get current time and remove this flag.